src/node.rs

```diff
@@ -79,19 +77,22 @@ impl NodeBuilder {
     /// request a new name and integrate itself into the network using the new name.
     ///
     /// The initial `Node` object will have newly generated keys.
-    pub fn create(self) -> Result<Node, RoutingError> {
-        let mut ev_buffer = EventBuf::new();
-
+    pub fn create(self) -> Result<(Node, mpmc::Receiver<Event>), RoutingError> {
         // start the handler for routing without a restriction to become a full node
-        let (_, machine) = self.make_state_machine(&mut ev_buffer);
-        let (tx, rx) = mpsc::channel();
+        let (interface_result_tx, interface_result_rx) = mpsc::channel();
+        let (user_event_tx, user_event_rx) = mpmc::unbounded();
+
+        let (_, machine) = self.make_state_machine(&mut user_event_tx.clone());
```

Comment:
This looks weird to me. Do we need to clone `user_event_tx` here? The `StateMachine` doesn't seem to be storing the reference anywhere, so the borrow should be over after this call, no?

Reply:
Yes, good catch :+1: